pkg/v3/coordinator/coordinator.go

```diff
@@ -234,7 +249,8 @@ func (c *coordinator) Start(pctx context.Context) error {
 		return fmt.Errorf("process already running")
 	}
 
-	go c.cacheCleaner.Run(c.cache)
+	go c.cacheCleaner.Start(c.cache)
+	go c.visitedCleaner.Start(c.visited)
```

Comment:
Would it make sense to have the cleaner as part of the cache itself? It's easy to forget to run a cache cleaner with this approach and cause memory leaks, and as far as I'm aware, the majority, if not all caches that we use, have an associated cleaner that needs to be started separately, outside the cache creation?